src/lib/proof-system/zkprogram.ts

```diff
@@ -161,12 +161,22 @@ const FeatureFlags = {
 };
 
 function createProgramState() {
+  let auxInputCache: Map<string, any[]> = new Map();
```

Comment:
I think we should rename this to avoid confusion. The methods below already handle axuililary output, if we name this auxiliary too it might cause some confusion